docs/diagrams/ModelClassDiagram.puml

```diff
@@ -5,46 +5,50 @@ skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
 Package Model <<Rectangle>>{
-Class "<<interface>>\nReadOnlyAddressBook" as ReadOnlyAddressBook
+Class "<<interface>>\nReadOnlyApplicationBook" as ReadOnlyApplicationBook
 Class "<<interface>>\nReadOnlyUserPrefs" as ReadOnlyUserPrefs
 Class "<<interface>>\nModel" as Model
-Class AddressBook
+Class ApplicationBook
 Class ModelManager
 Class UserPrefs
 
-Class UniquePersonList
-Class Person
-Class Address
+Class UniqueApplicationList
+Class Application
+Class Company
+Class Position
 Class Email
-Class Name
-Class Phone
+Class Date
+Class Contact
 Class Tag
+Class Interview
 
 }
 
 Class HiddenOutside #FFFFFF
 HiddenOutside ..> Model
 
-AddressBook .up.|> ReadOnlyAddressBook
+ApplicationBook .up.|> ReadOnlyApplicationBook
 
 ModelManager .up.|> Model
 Model .right.> ReadOnlyUserPrefs
-Model .left.> ReadOnlyAddressBook
-ModelManager -left-> "1" AddressBook
+Model .left.> ReadOnlyApplicationBook
+ModelManager -left-> "1" ApplicationBook
 ModelManager -right-> "1" UserPrefs
 UserPrefs .up.|> ReadOnlyUserPrefs
 
-AddressBook *--> "1" UniquePersonList
-UniquePersonList --> "~* all" Person
-Person *--> Name
-Person *--> Phone
-Person *--> Email
-Person *--> Address
-Person *--> "*" Tag
-
-Name -[hidden]right-> Phone
-Phone -[hidden]right-> Address
-Address -[hidden]right-> Email
-
-ModelManager -->"~* filtered" Person
+ApplicationBook *--> "1" UniqueApplicationList
+UniqueApplicationList -left-> "~* all" Application
+Application *--> Company
+Application *--> Contact
+Application *--> Email
+Application *--> Position
+Application *--> Date
```

Comment:
there seems to be a mistake when rendering this as a diagram (date is detached from everything else)?